### PR TITLE
Update install.sh to include the dependencies of diff.py for --extra

### PR DESCRIPTION
**Pull request by Gorialis**
`install.sh` doesn't install certain dependencies like `colorama`, even though it does install other ones:
https://github.com/Gorialis/papermario/blob/1f54aaefa1605837375e8f91ef792294b88a18a1/install.sh#L8
https://github.com/Gorialis/papermario/blob/1f54aaefa1605837375e8f91ef792294b88a18a1/install.sh#L13

However, `colorama` and others are required by `diff.py` and it will fail without them:
https://github.com/Gorialis/papermario/blob/1f54aaefa1605837375e8f91ef792294b88a18a1/diff.py#L218-L228

This adjusts install.sh so `install.sh --extra` will install the extra `diff.py` dependencies.
This also pulls the dependencies out of `install.sh` into their own `requirements.txt` and `requirements_extra.txt` so separate distros will not have to maintain their own Python dependencies in the future.

### CODE REVIEW

**Review by pixel-stuck — COMMENTED** [review]
This looks good to me. Holding off on approving, as I'm curious if we should keep install.sh and the associated dependency files in the root directory of the repo. install.sh is usually only used once when bootstrapping a new repo, and tool-wise, we usually reserve the root for tools which are used often. Additionally, the dependency files add to the clutter in the root. I'll wait for Ethan to discuss.

**Comment by Gorialis** [comment]
I've added `pycparser` and `PyYAML` to the main dependencies, as these are required by n64splat in `make setup`.

**Review by nanaian — APPROVED** [review]
LGTM; using `requirements.txt` was a good call.

@pixel-stuck I'm fine with moving `install.sh` and the requirements files to `tools/` (even though it isn't really a _tool_ per se). I personally see it as out-of-scope for this PR but if @Gorialis wants to move the files, make sure to update the relevant lines in the README.

**Comment by nanaian** [comment]
Relevant discussion:
image

**Pull request merged**
